app/server/appsmith-server/src/main/java/com/appsmith/server/services/NewActionServiceImpl.java

```diff
@@ -862,6 +862,30 @@ private DatasourceConfiguration updateDatasourceConfigurationForPagination(Actio
         return repository.findByPageId(pageId);
     }
 
+    @Override
+    public Mono<Boolean> setOnLoad(List<ActionDTO> actions) {
+        if (actions == null) {
+            return Mono.just(Boolean.FALSE);
+        }
+
+        List<ActionDTO> toUpdateActions = new ArrayList<>();
+
+        for (ActionDTO action : actions) {
+            // If a user has ever set execute on load, this field can not be changed automatically. It has to be
+            // explicitly changed by the user again. Add the action to update only if this condition is false.
+            if (!TRUE.equals(action.getUserSetOnLoad())) {
+                action.setExecuteOnLoad(TRUE);
+                toUpdateActions.add(action);
+            }
+        }
+
+        return Mono.just(toUpdateActions)
+                .flatMapMany(Flux::fromIterable)
+                .flatMap(actionDTO -> updateUnpublishedAction(actionDTO.getId(), actionDTO))
+                .collectList()
+                .thenReturn(TRUE);
```

Comment:
I think we can simplify this code.

```suggestion
return Flux.fromIterable(toUpdateActions)
                .flatMap(actionDTO -> updateUnpublishedAction(actionDTO.getId(), actionDTO))
                .then(Mono.just(TRUE));```